src/Auth0.AspNetCore.Authentication/Auth0WebAppWithAccessTokenAuthenticationBuilder.cs

```diff
@@ -194,9 +194,9 @@ private void EnableWithAccessToken()
 
         private static void ValidateOptions(Auth0WebAppOptions options)
         {
-            if (string.IsNullOrWhiteSpace(options.ClientSecret))
+            if (string.IsNullOrWhiteSpace(options.ClientSecret) && options.ClientAssertionSecurityKey == null)
             {
-                throw new ArgumentNullException(nameof(options.ClientSecret), "Client Secret can not be null when requesting an access token.");
+                throw new InvalidOperationException("Both Client Secret and Client Assertion can not be null when requesting an access token, one or the other has to be set.");
```

Comment:
As it's mo longer just an ArgumentNullException, I changed it to an InvalidOperation, as it's not valid to try and use access tokens without any of those two set.